# **Neural Networks implemented in Python's Tensor flow**

## **Part 3, our First Neural Networks:**

### **Section 1, :**

    Lets import our stuff again

In [8]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

import tensorflow as tf
from tensorflow import keras
from tensorflow.keras import layers # type: ignore
import tensorflow.compat.v2.feature_column as fc # type: ignore
import tensorflow.compat.v1 as tf1 # type: ignore

    Our dataset for today is the fassion dataset from keras, imported as follows

In [12]:
fashion_mnist = keras.datasets.fashion_mnist
(train_images, train_labels), (test_images, test_labels) = fashion_mnist.load_data()  # split test & train
train_images.shape

(60000, 28, 28)

    As we can see our dataset appears to be 60,000 images of 28*28 pixels
    So lets snoop 1 of these pixels

In [13]:
train_images[0,23,23]

194

    Hmm, 194? that doesn't mean much, lets try this:

In [32]:
print(train_images[15:16])

[[[  0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
     0   0   0   0   0   0   0   0   0   0   0]
  [  0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
     0   0   0   0   0   0   0   0   0   0   0]
  [  0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
     0   0   0   0   0   0   0   0   0   0   0]
  [  0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
     0   0   0   0   0   0   0   0   0   0   0]
  [  0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
     0   0   0   0   0   0   0   0   0   0   0]
  [  0   0   0   0   0   0   0   0   0   0   0   0   0   0  85 188 146
    79   0   0   0   0   0   0   0   0   0   0]
  [  0   0   0   0   0   0   0   0   0   0   0   0   0   0 140 202 199
   255 144   0   0   0   0  11 135 157   5   0]
  [  0   0   0   0   0   0   0   0   0   0   0   3   0   0 200 187 200
   191 255  51   0   0  16 208 227 236  63   0]
  [  0   0   0   0   0   0   0   0   0   0   0  

    so from this we can conclude each one is a B/W pixel from 0 to 255 - & the edges have white pixels